api.go

```diff
@@ -40,37 +40,78 @@ func (b *Bot) Raw(method string, payload interface{}) ([]byte, error) {
 	return json, nil
 }
 
+func addFileToWriter(writer *multipart.Writer, fieldName string, file interface{}) error {
+	var reader io.Reader
+	var part io.Writer
+	var err error
+	var filename string
+
+	if r, ok := file.(io.Reader); ok {
+		// Telegram requires fields to have a filename, otherwise
+		// `Bad Request: wrong URL host` would be returned
+		filename = "empty"
```

Comment:
This led to make uploaded file always named "empty", despite FileName variable :(